### PR TITLE
Browser-related functionality changes (close #66)

### LINKED ISSUE #66: Get rid of the old browser inactivity detection mechanism

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @inikulin @helen-dikareva

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
FPR (tests fixed)

**Comment by helen-dikareva** [comment]
lgtm

File: src/client/browser/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import COMMAND from '../../browser-connection/command';
+import hammerheadApi from './deps/hammerhead';
+
+
+const HEARTBEAT_INTERVAL = 30 * 1000;
+
+
+//Utils
+var XMLHttpRequest = hammerheadApi ? hammerheadApi.NativeMethods.XMLHttpRequest : window.XMLHttpRequest;
```
Comment (inikulin) [inline src/client/browser/index.js]: Can we avoid doing it, e.g. by passing XMLHttpRequest ctor to methods (with appropriate comment which clarifies why we do this)? I don't like hammerhead dep here as well.

File: src/client/browser/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+function sendXHR (url, callback) {
+    var xhr = new XMLHttpRequest();
```
Comment (inikulin) [inline src/client/browser/index.js]: Why not Promise?

File: src/client/core/transport.js [hunk 3]
```diff
@@ -1,81 +1,25 @@
 import * as hammerheadAPI from './deps/hammerhead';
```
Comment (inikulin) [inline src/client/core/transport.js]: Here and in all other places: why it's called hammerheadAPI why not just hammerhead?

**Comment by inikulin** [comment]
\r-

File: src/client/browser/index.js (hunk 1 above)
Comment (AlexanderMoskovkin) [inline src/client/browser/index.js]: done

File: src/client/browser/index.js (hunk 2 above)
Comment (AlexanderMoskovkin) [inline src/client/browser/index.js]: done. We use the `es6-promise` module on the client-side and the `promise` module on the server-side. We should switch to the `es6-promise` everywhere (like in the `testcafe-hammerhead` repo), but not in this pr

File: src/client/core/transport.js (hunk 3 above)
Comment (AlexanderMoskovkin) [inline src/client/core/transport.js]: Agree, we should use `hammerhead`. But this refactoring will affect more than 30 files. Let's fix it in the https://github.com/superroma/testcafe-phoenix/issues/70 issue

**Comment by AlexanderMoskovkin** [comment]
FPR

File: src/client/browser/index.js (hunk 2 above)
Comment (inikulin) [inline src/client/browser/index.js]: Let's create a separate issue then

File: src/client/browser/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        };
+
+        xhr.send(null);
+    });
+}
+
+
+//API
+export function startHeartbeat (heartbeatUrl, XMLHttpRequest) {
+    sendXHR(heartbeatUrl);
```
Comment (inikulin) [inline src/client/browser/index.js]: We should pass XMLHttpRequest request here as well.

File: src/client/browser/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { Promise } from 'es6-promise';
+import COMMAND from '../../browser-connection/command';
+
+
+const HEARTBEAT_INTERVAL = 30 * 1000;
+
+
+//Utils
+function sendXHR (url, XMLHttpRequest) {
```
Comment (inikulin) [inline src/client/browser/index.js]: We need a comment why we pass XMLHttpRequest here.

**Comment by inikulin** [comment]
\r0

File: src/client/browser/index.js (hunk 2 above)
Comment (AlexanderMoskovkin) [inline src/client/browser/index.js]: https://github.com/superroma/testcafe-phoenix/issues/71

**Comment by AlexanderMoskovkin** [comment]
FPR

File: src/client/browser/index.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+// the original URL. That's why we need the XMLHttpRequest argument to send the request via native methods.
+function sendXHR (url, XMLHttpRequest) {
+    return new Promise((resolve) => {
+        var xhr = new XMLHttpRequest();
+
+        xhr.open('GET', url, true);
+
+        xhr.onreadystatechange = () => {
+            if (xhr.readyState === 4 && xhr.status === 200)
+                resolve(JSON.parse(xhr.responseText));
```
Comment (AlexanderMoskovkin) [inline src/client/browser/index.js]: This fails when response is empty. I'll fix it soon

**Comment by inikulin** [comment]
lgtm

**Comment by inikulin** [comment]
> This fails when the response is empty. I'll fix it soon
\r- then =)

File: src/browser-connection/local.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ export default class LocalBrowserConnection extends BrowserConnection {
         super(gateway);
 
         this.forceClose = false;
+        this.forceIdle  = false;
```
Comment (inikulin) [inline src/browser-connection/local.js]: It's should be `switchedToIdleOnClose` since it represents the *state*.

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by inikulin** [comment]
\r-, see comment above

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
thanks

**Pull request merged**